### PR TITLE
[OAP-1846][oap-native-sql] add unit tests from spark to native sql

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

add some unit tests from spark to native sql
https://github.com/Intel-bigdata/OAP/issues/1846

### CODE REVIEW

**Comment by zhztheplayer** [comment]
Do we have to copy/paste all these codes from Spark code base. I mean it will be an option to import them automatically on compile time using maven plugins like  `exec-maven-plugin` or `maven-resource-plugin` etc.

**Comment by rui-mo** [comment]
@zhztheplayer In these tests, we changed the spark conf to use native sql engine, and ignored the current failing ones. If importing automatically, seems vanilla spark will be used in testing?

File: oap-native-sql/core/src/test/scala/org/apache/spark/sql/ApproxCountDistinctForIntervalsQuerySuite.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      .set("spark.sql.execution.arrow.maxRecordsPerBatch", "4096")
+      //.set("spark.shuffle.manager", "org.apache.spark.shuffle.sort.ColumnarShuffleManager")
+      .set("spark.memory.offHeap.enabled", "true")
+      .set("spark.memory.offHeap.size", "10m")
+      .set("spark.sql.join.preferSortMergeJoin", "false")
+      .set("spark.sql.columnar.codegen.hashAggregate", "false")
+      .set("spark.oap.sql.columnar.wholestagecodegen", "false")
+      .set("spark.sql.columnar.window", "false")
+      .set("spark.unsafe.exceptionOnMemoryLeak", "false")
+      .set("spark.sql.columnar.tmp_dir", "/codegen/nativesql/")
```
Comment (zhouyuan) [inline oap-native-sql/core/src/test/scala/org/apache/spark/sql/ApproxCountDistinctForIntervalsQuerySuite.scala:45]: should use runtime created tmp dir, travis tests is not able to touch dirs under '/'

**Comment by zhouyuan** [comment]
@zhztheplayer trying to place the base layout of unit/functional tests for native engine here. currently only a small part of tests are migrated to the codebase - due to the resource limitation of Travis CI, most of the big tests will be moved to local Jenkins cluster.  

related: https://github.com/Intel-bigdata/OAP/issues/1646

**Comment by zhouyuan** [comment]
@rui-mo can you please do a rebase on master? Travis should be fixed now

**Comment by rui-mo** [comment]
@zhouyuan it was reabsed to master already, but seems the Travis still failed in mllib.

**Comment by zhouyuan** [comment]
notes: the logic in travis test script is to bypass https://github.com/Intel-bigdata/OAP/issues/1745. we should clean up this once #1745 is fixed

**Pull request merged**